tests/test_bigquery_eager.py

```diff
@@ -251,6 +283,45 @@ def tearDownClass(cls):  # pylint: disable=invalid-name
         """setUpClass"""
         cls.server.stop()
 
+    def _get_read_session(self, client):
+        return client.read_session(
+            self.PARENT,
+            self.GCP_PROJECT_ID,
+            self.TABLE_ID,
+            self.DATASET_ID,
+            selected_fields=[
+                "string",
+                "boolean",
+                "int",
+                "long",
+                "float",
+                "double",
+                "repeated_string",
+                "repeated_double",
+            ],
+            selected_fields_repeated=[
```

Comment:
I don't think that such API is easy to use. Consider case when you need to select hundreds of columns, and some of them are repeated. Maintaining such code would be a problem.

Could you instead change API so if selected fields value ends with [], field is considered repeated.
So in this case it is going to look like:
            selected_fields=[
                "string",
                "boolean",
                "int",
                "long",
                "float",
                "double",
                "repeated_string[]",
                "repeated_double[]",
            ],

Reply:
And could you also add a comment to https://github.com/tensorflow/io/blob/master/tensorflow_io/bigquery/README.md explaining how to select repeated fields.

Reply:
I agree with you.  The reason I put selected_fields_repeated in a current way, as you may already realize, is to align existing pattern of selected_fields and output_types. I have concerns around this as well, so want to make sure we agree with API first before i proceeding furthur (adding other primitive types, adding more comprehensive tests, add docs etc). 

I don't think 
```
"repeated_string[]",
"repeated_double[]",
```
is a good idea. We are using `[]` trick, which is unintuitive, to represent the semantic meaning of `is repeated`. 

Ideally,  selected_fields, output_types, selected_fields_repeated, these three should be represented as a dictionary
{
 'field': a_selected_field,  'mode': repeated_or_not,  'output_types': a_output_type,
...
}
right now these three are three lists, we rely on the index to list to reach that field:mode:output_type mapping semantic.

Reply:
the reason I didn't choose the route of making these three in the dictionary in this pull request directly primarily makes the API backwards compatible.

Reply:
I like your proposal.
How about changing API to accept selected_fields as either array, as it is now or as a dictionary, the way you suggested?
If it is specified as a dictionary, ouput_type should not be specified.

And another question. Is it actually necessary to know whether field is repeated or not on the python side?
Would it be possible to just derive that information in C++ from AVRO schema?

Reply:
> How about changing API to accept selected_fields as either array, as it is now or as a dictionary, the way you suggested?

Sounds good. I will make the change.

> And another question. Is it actually necessary to know whether field is repeated or not on the python side?
Would it be possible to just derive that information in C++ from AVRO schema?

Good question. I gave a try a while ago, the AVRO schema from C++ do contain the information but it is a Tensor. I am not sure having a session run over there is a good idea (maybe i will give anther stab later to see how to best hide *repeated* from user). Open to suggestions :) .  Just sharing, internally we build a higher-level abstraction, say BigQueryDataSet, which calls BigQuery Python Client API to get the Avro Schema and then call this tfio API so user doesn't need to worry about setting repeated at all.

Reply:
Update: It seems I couldn't find a very elegant way at this moment to avoid treating repeated as a user input parameter for now. This repeatd_field information is used in tensorflow_io/bigquery/python/ops/bigquery_api.py to set shape for tensors in self._element_spec. 

I have updated the code to be the API as we discussed above: support both list and dict for selected_fields